imagenameparser/api/image_name.py

```diff
@@ -0,0 +1,896 @@
+    values that can be used as substitution keywords in the Mash Job doc
+    fragment file content, as returned by the ``substitutions` property.
+
+    Finally, ``fragment_files`` is a property provides that provides a
+    list of fragment files, located under the ``mashhouse/data/fragments``
+    tree that should be loaded, in the order specified, to construct the
+    required Mash Job doc for the specified framework & name combination.
+
+    Fragment Files list::
+    The list of fragment files implements a depth first traversal of the
+    fragment hierarchy, with the content of subsequent fragments taking
+    precedence over previous versions. At each level there can be ``byos``
+    and ``payg`` directories containing payment model specific entries.
+
+    Additionally when the traversal reaches the release version hierarchy
+    we will load the fragments for earlier service packs or minor releases
+    in the same version stream in sequence until we reach the targetted
+    release. This allows relevant settings from earlier releases to be
+    inherited if there are no specific settings for the target release.
+
+    Note that when the list of candidate fragment files is traversed,
+    only those fragment files that exist will be loaded; the list will
+    contain all possible fragment files that might need to be loaded
+    but in reality only a subset are usually needed due to inheritance.
+    """
+
+    def __init__(self, framework, name):
+        self._framework = framework
+        self._image_name = name
+        self._image_info = None
+        self._fragment_files = None
+        self._image_job = None
+
+        self._extract_image_info()
+
+    def _extract_image_info(self):
+        """Extract the required image details from the image name
+
+        Raises:
+            BadRegexMatchError: If the image name does not match the expected
+                regex pattern
+        """
+        if self._image_info is None:
+            matched = ImageNameRegexp().match(self._image_name)
+            try:
+                self._image_info = matched.groupdict()
+            except AttributeError:
+                raise BadRegexMatchError("Could not match regex for image: "
+                                         + self._image_name) from None
+            if self._framework in ("azure", "ec2") and \
+                    self._image_info['sle_server'] is not None and \
+                    self._image_info['azure_hosted'] is None:
+                if self._image_info['prodbase'] != "suse-sles":
+                    raise BadRegexMatchError(self._framework + " image "
+                                             + self._image_name
+                                             + " does not start with 'suse'")\
+                        from None
+
+            _logger.debug("Image match values for image %s are: %s",
+                          self._image_name, self._image_info)
+
+    @property
```

Comment:
Suspect you don't need the substitutions property which is intended to be used as an input to a templating framework?